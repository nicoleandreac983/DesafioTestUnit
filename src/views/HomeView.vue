<script>

</script>

<template>
   <div>
    <h1 class="Texto">Homeview</h1>
    <button class="boton">Enviar</button>
  </div>
</template>

<style scoped>  
.Texto {
    color: blue;
    font-size: 24px;
}
</style>
