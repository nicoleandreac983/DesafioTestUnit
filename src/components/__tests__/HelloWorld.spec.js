import { describe, test, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import HomeView from '@/views/HomeView.vue'
import { createRouter, createWebHistory } from "vue-router";

describe("HomeView", () => {
  test("Probando la existencia del componente o vista HomeView", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/home",
          name: "home",
          component: HomeView,
        },
      ],
    });
 
    router.push("/home");
    await router.isReady();
 
    const wrapper = mount(HomeView, {
      global: {
        plugins: [router],
      },
    });
 
    expect(wrapper.findComponent(HomeView).exists()).toBe(true);
  });
});

describe('Componente HomeView.vue', () => {
  test('Validacion de match con el snapshot', () => {
    const wrapper = shallowMount(HomeView)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

// snapshot de la estructura HTML del componente HomeView.vue

describe('Componente HomeView.vue', () => {
  test('Validar el texto de un botÃ³n de enviar formulario', () => {
    const wrapper = shallowMount(HomeView)
    const botonEnviar = wrapper.find('.boton')
    console.log(botonEnviar.text())
    expect(botonEnviar.text()).toBe('Enviar')
  })
})
