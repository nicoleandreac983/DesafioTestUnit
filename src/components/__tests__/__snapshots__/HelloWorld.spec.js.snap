// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente HomeView.vue > Validacion de match con el snapshot 1`] = `
"<div data-v-b4e148ca="">
  <h1 data-v-b4e148ca="" class="Texto">Homeview</h1><button data-v-b4e148ca="" class="boton">Enviar</button>
</div>"
`;
